### PR TITLE
fix(v2): fix double empty text node + initial qwik city fixes

### PR DESCRIPTION
**Pull request by Varixo**
This PR fixes double empty text node rendering problem and adds initial qwik city fixes

### CODE REVIEW

File: packages/qwik/src/core/v2/client/dom-container.ts [hunk 1]
```diff
@@ -87,18 +87,12 @@ export function getDomContainerFromHTMLElement(qContainerElement: Element): ICli
...
     ? (vnode_getDomParent(element) as Element)
     : element;
-  while (
-    qContainerElement &&
-    (!qContainerElement.hasAttribute(QContainerAttr) ||
-      isQContainerElementWithValue(qContainerElement))
-  ) {
-    qContainerElement = qContainerElement.closest(QContainerSelector);
-  }
-  return qContainerElement;
```
Comment (Varixo) [inline packages/qwik/src/core/v2/client/dom-container.ts:101]: not sure why this while loop was here

Comment (mhevery) [reply packages/qwik/src/core/v2/client/dom-container.ts:101]: no idea

File: packages/qwik/src/core/index.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export type { QwikDOMAttributes, QwikJSX, QwikJSX as JSX } from './render/jsx/ty
 export type { QwikIntrinsicElements } from './render/jsx/types/jsx-qwik-elements';
 export type { QwikHTMLElements, QwikSVGElements } from './render/jsx/types/jsx-generated';
 export { render2 as render } from './v2/client/dom-render';
-export { getDomContainer } from './v2/client/dom-container';
+export { getDomContainer, getQContainerElement } from './v2/client/dom-container';
```
Comment (mhevery) [inline packages/qwik/src/core/index.ts:84]: You are making `getQContainerElement` public. I am not sure we should do that. We will need to document it and support it. I think it is better for it to stay private.

**Review by mhevery — CHANGES_REQUESTED** [review]

File: packages/qwik/src/core/index.ts (hunk 2 above)
Comment (Varixo) [reply packages/qwik/src/core/index.ts]: I have changed `getQContainerElement` from public to internal. It is okay?

**Review by mhevery — APPROVED** [review]

**Pull request merged**